src/Honeycomb/Api/Events.hs

```diff
@@ -18,13 +19,13 @@ import Network.URI (normalizeEscape)
 
 sendEvents ::
   MonadIO m =>
-  Client.Manager ->
+  (Client.Request -> m (Client.Response LBS.ByteString)) ->
```

Comment:
How about adding an opaque type like `Client` (HC API specific) that hides details of this function?
i think that'll look nicer on the type level and be more source-compatible with whatever future tweaks you'll want to make to bootstrap process.

Code changes will be minimal too, instead of `Client.Manager` we'll just have `HT.Client`.

WDYT?

Reply:
I don't think it would be completely "opaque", as it needs to be modifiable (at least through lenses) in test code. But yes, good idea.

Reply:
I could imagine (later) adding implementations for uuid and random.

Reply:
Hmm. Actually my thought is about a Client type at the user-facing end of the API. At the point where one calls `sendEvents`, I question the use of a heavy object, which may have grown more tentacles, and therefore be more complex to mock than needed. So I may leave this type here, but have a larger type at library initialisation?

Reply:
... hmm though in a sense, this is captured by `HoneyServerOptions`. Not sure we gain much by splitting the call out from there.

Reply:
Sure, I guess if you don't see this as a "user facing" API - point is moot, it's a private implementation detail then.